beanquery/shell.py

```diff
@@ -401,7 +401,7 @@ def _extract_queries(self, entries):
     def do_errors(self, arg=None):
         "Print the errors that occurred during Beancount input file parsing."
         if self.context.errors:
-            printer.print_errors(self.context.errors)
+            printer.print_errors(self.context.errors, file=sys.stderr)
```

Comment:
Is this about the output of the shell's `.errors` command or about the errors printed when a ledger is loaded? In the first case, I think the output of commands should always be directed to stdout. In the latter case, the fix affects also the output of the `.errors` command, which should still go to stdout. In both cases, having the errors go to stderr but the string `(no errors)` that is printed when there are no errors to display going to stdout does not seem consistent.

Reply:
It's about the errors printed when a ledger is loaded. In the current state there is a regression because you can't pipe the output of `bean-query -f csv ...` when there are errors in the ledger. With previous version you can do a `bean-query -f csv ... 2>/dev/null` and move on.

I didn't know about the `.errors` command. I'll check it and propose another fix.

Thanks for the review.